swap_meet/vendor.py

```diff
@@ -1,2 +1,160 @@
+        Swaps best items in provided categories between two instances' inventories and returns True.
+        - Returns False if either item not found. 
+
+    get_newest_item():
+        Returns newest item from instance's inventory items with known ages.
+        - Returns None if all items ages are unknown.
+
+    swap_by_newest(other):
+        Swaps newest items of two instances' inventories and returns True.
+        - Returns False if either or both inventory items' ages are all unknown.
+    
+    """
+
+    def __init__(self, inventory = None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+    
+    def add(self, new_item):
+        """Add provided item to inventory"""
+
+        self.inventory.append(new_item)
+        return new_item
+    
+    def remove(self, discarded_item):
+        """Remove an item from inventory"""
+
+        if discarded_item not in self.inventory:
+            return False
+        else: 
+            self.inventory.remove(discarded_item)
+            return discarded_item
+    
+    def get_by_category(self, category_string):
+        """Return list of items from inventory of a certain category"""
+
+        item_list = []
+        for item in self.inventory:
+            if item.category == category_string:
+                item_list.append(item)
+        return item_list
+    
+    def swap_items(self, other, my_item, their_item):
+        """Swaps items betwen self and another vendor's inventory"""
+
+        # check valid input
+        if my_item not in self.inventory or their_item not in other.inventory:
+            return False
+        else:
+            # remove items from original inventory
+            self.remove(my_item)
+            other.remove(their_item)
+            # add items to new inventory 
+            self.add(their_item)
+            other.add(my_item)
+            return True
+    
+    def swap_first_item(self, other):
+        """Swaps first item in self and another vendor's inventory"""
+
+        # check valid input
+        if not self.inventory or not other.inventory:
+            return False
+        else:
+            my_first_item = self.inventory[0]
+            vendor_first_item = other.inventory[0]
+            self.swap_items(other, my_first_item, vendor_first_item)
+            return True
+    
+    def get_best_by_category(self, desired_category):
+        """Returns item in best condition within a certain category"""
+
+        # get list of items of category
+        potential_items = self.get_by_category(desired_category)
+        # account for no items of desired category or only one of desired category
+        if not potential_items:
+            return None
+        elif len(potential_items) == 1:
+            return potential_items[0]
+        # calculate max item condition
+        else:
+            best_item = max(potential_items, key = lambda item: item.condition)
+            return best_item
```

Comment:
I'm not sure we need to separate out the case where the list only contains one item. What do we gain by handling the logic for a single item in the list separately from lists of 2 or more items?

If we can combine cases, we can simplify the code a little:
```suggestion
        if not potential_items:
            return None
            
        best_item = max(potential_items, key = lambda item: item.condition)
        return best_item
```

Reply:
I think I was figuring if there was just the one item it would save time (and maybe space as well) to not run a more complicated calculation to figure out the max - with only one option, it is by default the max. But perhaps that's not the case, depending on how the calculation works? Or maybe it just doesn't save enough time to make it worth it.

Reply:
That's a great consideration to have, I think it'd be worth writing a short comment to that effect above the `elif` to bring a little clarity. In this project we don't need the fast access if there's a single item, but there are many applications where it's important to be as performant as possible, and we may absolutely want to do something like this.